okio/src/test/java/okio/BufferedSinkTest.java

```diff
@@ -211,4 +212,41 @@
     } catch (ArrayIndexOutOfBoundsException expected) {
     }
   }
+
+  @Test public void longDecimalString() throws IOException {
+    assertLongDecimalString(0);
+    assertLongDecimalString(Long.MIN_VALUE);
+    assertLongDecimalString(Long.MAX_VALUE);
+
+    for (int i = 1; i < 20; i++) {
+      long value = BigInteger.valueOf(10L).pow(i).longValue();
+      assertLongDecimalString(value - 1);
+      assertLongDecimalString(value);
+    }
+  }
+
+  private void assertLongDecimalString(long value) throws IOException {
```

Comment:
Confirm the buffer is left in the expected state? Maybe write a trailing `abc` and read it back?